premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs

```diff
@@ -56,16 +58,27 @@ public class PreEnrollmentService
 
         var student = await _studentValidationService.ValidateStudentExists(studentEmail);
 
-        if( !preEnrollment.CanBeChangedByStudent(student))
+        if (!preEnrollment.CanBeChangedByStudent(student))
             throw new InvalidPreEnrollmentSelectionException("Student cannot change PreEnrollment");
 
         if (preEnrollment.Selections.Count < courseOfferings.Length)
             throw new InvalidPreEnrollmentSelectionException("Cannot remove more selections than what are available");
-        
+
         preEnrollment.RemoveSelections(courseOfferings);
         _preEnrollmentRepository.Save(preEnrollment);
     }
 
+    public async Task<PreEnrollment> CreateNewPreEnrollment(string studentEmail, string name, int semesterId)
+    {
+        var student = await _studentValidationService.ValidateStudentExists(studentEmail);
+        await _semesterValidationService.ValidateSemesterExists(semesterId);
+        if (await _preEnrollmentRepository.ContainsWithNameStudentAndSemesterId(name, student.Id, semesterId))
+            throw new CoreException("PreEnrollment Already Exists");
```

Comment:
Maybe we could eventually say "PreEnrollment with this name already exists" that way the user knows what needs to change so he can create a PreEnrollment